engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/BusinessRuleTaskTest.java

```diff
@@ -205,4 +209,129 @@ public void shouldUseResultInOutputMappings() {
         .extracting(VariableRecordValue::getScopeKey, VariableRecordValue::getValue)
         .containsExactly(processInstanceKey, "\"Jedi\"");
   }
+
+  @Test
+  public void shouldWriteDecisionEvaluationEvent() {
+    // given
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlClasspathResource(DMN_RESOURCE)
+            .withXmlResource(
+                processWithBusinessRuleTask(
+                    t ->
+                        t.zeebeCalledDecisionId("force_user").zeebeResultVariable(RESULT_VARIABLE)))
+            .deploy();
+
+    final var calledDecision =
+        deployment.getValue().getDecisionsMetadata().stream()
+            .filter(decision -> decision.getDecisionId().equals("force_user"))
+            .findFirst()
+            .orElseThrow();
+
+    // when
+    final Map<String, Object> variables =
+        Map.ofEntries(entry("lightsaberColor", "blue"), entry("height", 182));
+    final long processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).withVariables(variables).create();
+
+    final var businessRuleTaskActivated =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .withElementType(BpmnElementType.BUSINESS_RULE_TASK)
+            .getFirst();
```

Comment:
❓ Why not continue using `findFirst().orElseThrow()`?

Reply:
Different context. `findFirst().orElseThrow()` is used on a Java stream that is created from a deployment event but not from a `RecordStream`.

I agree that `findFirst().orElseThrow()` is in general more explicit but we already use `getFirst()` in other places :sweat_smile: